src/main/kotlin/org/rust/lang/core/types/Extensions.kt

```diff
@@ -78,7 +78,12 @@ fun Ty.builtinDeref(explicit: Boolean = true): Pair<Ty, Mutability>? =
         else -> null
     }
 
-val RsExpr.isDereference: Boolean get() = this is RsUnaryExpr && this.mul != null
-
+val RsUnaryExpr.isDereference: Boolean get() = this.mul != null
 
 val RsExpr.isMutable: Boolean get() = mutabilityCategory?.isMutable ?: Mutability.DEFAULT_MUTABILITY.isMut
+
+tailrec fun unwrapParenExprs(expr: RsExpr): RsExpr =
+    if (expr is RsParenExpr) unwrapParenExprs(expr.expr) else expr
+
+val RsExpr.isAssignBinaryExpr: Boolean
+    get() = this is RsBinaryExpr && this.operatorType is AssignmentOp
```

Comment:
Let's move these functions to `RsExpr.kt`.
`org.rust.lang.core.types.Extensions.kt` is for extending PSI with types. All simple PSI extensions should be located inside `org.rust.lang.core.psi.ext` package